core/proto/keytransparency_v1_grpc/keytransparency_v1_admin_grpc.proto

```diff
@@ -23,43 +23,63 @@ option go_package = "github.com/google/keytransparency/core/proto/keytransparenc
 // associated with it.
 package keytransparency.v1.grpc;
 
-import "proto/keytransparency_v1_proto/keytransparency_v1_admin_proto.proto";
 import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+import "proto/keytransparency_v1_proto/keytransparency_v1_admin_proto.proto";
 
-// The KeyTransparencyAdminService API provides batch access to the directory of public keys. 
+// The KeyTransparencyAdminService API provides the following resources:
+// - Delegated Actor
+//   - Batch updates to the directory of public keys, where authorized by users. 
+// - Domains 
+//   Namespaces on which which Key Transparency operates. A domain determines a
+//   unique Trillian map to use. It also determines the authentication policies
+//   for users and apps within a domain.
+//   - /v1/domains
 service KeyTransparencyAdminService {
-  // BatchSetEntries uses an authorized_public key to perform a set request on multiple entries at once.  
+  // BatchUpdateEntries uses an authorized_public key to perform a set request on multiple entries at once.  
   rpc BatchUpdateEntries(keytransparency.v1.proto.BatchUpdateEntriesRequest) returns (keytransparency.v1.proto.BatchUpdateEntriesResponse) {
     option (google.api.http) = {
       post: "/v1/users:batch"
       body: ""
     };
   }
 
+
+  //
+  // Domain Admin API
+  //
+
+  // ListDomains returns a list of all domains this Key Transparency server
+  // operates on.
   rpc ListDomains(keytransparency.v1.proto.ListDomainsRequest) returns (keytransparency.v1.proto.ListDomainsResponse) {}
 
   // GetDomain returns the confiuration information for a given domain.
   rpc GetDomain(keytransparency.v1.proto.GetDomainRequest) returns (keytransparency.v1.proto.GetDomainResponse) {
     option (google.api.http) = { get: "/v1/domains/{domain_id}" };
   }
 
+  // CreateDomain creates a new Trillian log/map pair.  A unique domainId must
+  // be provided.  To create a new domain with the same name as a previously
+  // deleted domain, a user must wait X days until the domain is garbage
+  // collected.
   rpc CreateDomain(keytransparency.v1.proto.CreateDomainRequest) returns (keytransparency.v1.proto.CreateDomainResponse) {
     option (google.api.http) = {
       post: "/v1/domains"
       body: "*"
     };
   }
 
-  rpc DeleteDomain(keytransparency.v1.proto.DeleteDomainRequest) returns (keytransparency.v1.proto.DeleteDomainResponse) {
+  // DeleteDomain marks a domain as deleted.  Domains will be garbage collected
+  // after X days.
```

Comment:
I assume the deletion interval X is outside of KT's control, an aspect of the Trillian deployment being used?

Reply:
Garbage collection is a too-be-implemented feature.  At the moment `X` is infinity. Once we have more than 1 domain, we can implement garbage collection at a frequency that matches Trillian.